### PR TITLE
s390x: Collect linuxrc logs in post_fail_hook

### PR DESCRIPTION
**Pull request by Soulofdestiny**
[related progress ticket](https://progress.opensuse.org/issues/25982)
[related MR for needles](https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/788)
[verification for z/VM](http://opeth.suse.de/tests/841)
[verification for (z)KVM](http://opeth.suse.de/tests/844)

### CODE REVIEW

**Comment by Soulofdestiny** [comment]
@okurz 
@rwx788 
WDYT? should help to find the root cause for issues before the installer is actually running

File: tests/installation/bootloader_s390.pm [hunk 1]
```diff
@@ -301,4 +302,25 @@ sub run {
...
+    my $s3270 = console('x3270');
+    my $r;
+
+    if (check_screen 'linuxrc') {
+        # Start linuxrc shell
+        $s3270->sequence_3270(qw(String("x") ENTER String("3") ENTER));
+        assert_screen 'linuxrc-shell';
+
+        # collect linuxrc logs
+        #$s3270->sequence_3270((String(\"cat /var/log/linuxrc.log\"), ENTER));
```
Comment (rwx788) [inline tests/installation/bootloader_s390.pm]: Let's remove this line

File: tests/installation/bootloader_s390.pm [hunk 2]
```diff
@@ -301,4 +302,25 @@ sub run {
...
+        $s3270->sequence_3270("String(\"cat /var/log/linuxrc.log && echo 'LINUXRC_LOG_SAVED'\")", "ENTER",);
+
+        $r = $s3270->expect_3270(
+            output_delim => qr/LINUXRC_LOG_SAVED/,
+            timeout      => 60
+        );
+        $r ? record_info 'Logs collected', 'Linuxrc logs can be found in autoinst-log.txt' : die "Could not save linuxrc logs";
```
Comment (rwx788) [inline tests/installation/bootloader_s390.pm]: I'n not sure how exactly it sequence_3270 works, but will it be in autoinst log, or in serial or in both?

File: tests/installation/bootloader_s390.pm [hunk 3]
```diff
@@ -301,4 +302,25 @@ sub run {
...
+    my $r;
+
+    if (check_screen 'linuxrc') {
+        # Start linuxrc shell
+        $s3270->sequence_3270(qw(String("x") ENTER String("3") ENTER));
+        assert_screen 'linuxrc-shell';
+
+        # collect linuxrc logs
+        #$s3270->sequence_3270((String(\"cat /var/log/linuxrc.log\"), ENTER));
+        $s3270->sequence_3270("String(\"cat /var/log/linuxrc.log && echo 'LINUXRC_LOG_SAVED'\")", "ENTER",);
```
Comment (rwx788) [inline tests/installation/bootloader_s390.pm]: I guess we don't need comma in the end

File: tests/installation/bootloader_s390.pm [hunk 4]
```diff
@@ -15,6 +15,7 @@
 use base "installbasetest";
 
 use testapi;
+use opensusebasetest;
```
Comment (okurz) [inline tests/installation/bootloader_s390.pm]: this module ISA installbasetest which ISA opensusebasetest so I do not get why this should be needed

File: tests/installation/bootloader_s390.pm [hunk 5]
```diff
@@ -301,4 +302,25 @@ sub run {
     $self->result('ok');
 }
 
+sub post_fail_hook {
+    my $s3270 = console('x3270');
+    my $r;
+
+    if (check_screen 'linuxrc') {
```
Comment (okurz) [inline tests/installation/bootloader_s390.pm]: meh, as you probably know by know I consider a check_screen with default timeout a bad practice but this time, being in the `post_fail_hook` … ok

File: tests/installation/bootloader_s390.pm (hunk 2 above)
Comment (okurz) [reply tests/installation/bootloader_s390.pm]: only in autoinst-log.txt

File: tests/installation/bootloader_zkvm.pm [hunk 6]
```diff
@@ -92,5 +92,21 @@ sub run {
...
+    type_line_svirt 'x';
+    wait_serial "Linuxrc extras";
+
+    # Start linuxrc shell
+    type_line_svirt '3';
+    wait_serial "ttysclp0:install";
+
+    # Collect Linuxrc logs
+    type_line_svirt "'cat /var/log/linuxrc.log > /dev/$serialdev && echo 'LINUXRC_LOG_SAVED' > /dev/$serialdev'";
+    wait_serial "LINUXRC_LOG_SAVED" ? record_info 'Logs collected', 'Linuxrc logs can be found in serial0.txt' : die "could not collect linuxrc logs";
```
Comment (okurz) [inline tests/installation/bootloader_zkvm.pm]: For all three type_line_svirt+wait_serial tuples: `type_line_svirt` has the "expect" parameter, see opensusebasetest::rewrite_static_svirt_network_configuration how it's used. So the section could/should be rewritten as:

```perl
    # Enter Linuxrc extra mode
    type_line_svirt 'x', expect => 'Linuxrc extras';

    # Start linuxrc shell
    type_line_svirt '3', expect => 'ttysclp0:install';

    # Collect Linuxrc logs
    type_line_svirt "'cat /var/log/linuxrc.log > /dev/$serialdev && echo 'LINUXRC_LOG_SAVED' > /dev/$serialdev'";
￼    wait_serial "LINUXRC_LOG_SAVED" ? record_info 'Logs collected', 'Linuxrc logs can be found in serial0.txt' : die "could not collect linuxrc logs";
```

**Comment by okurz** [comment]
needles merged

File: tests/installation/bootloader_zkvm.pm (hunk 6 above)
Comment (Soulofdestiny) [reply tests/installation/bootloader_zkvm.pm]: Nice. Didn't know that. Will change

**Comment by Soulofdestiny** [comment]
updated - will retrigger verification runs to be sure

**Comment by Soulofdestiny** [comment]
verifications:

http://opeth.suse.de/tests/855
http://opeth.suse.de/tests/853

**Pull request merged**